dojo/finding/views.py

```diff
@@ -435,207 +260,940 @@ def prefetch_for_similar_findings(findings):
+        }
+
+        return (
+            AcceptedFindingFilter(*args, **kwargs)
+            if self.get_filter_name() == "Accepted"
+            else FindingFilter(*args, **kwargs)
+        )
+
+    def get_filtered_findings(self):
+        findings = get_authorized_findings(Permissions.Finding_View).order_by(self.get_order_by())
+        findings = self.filter_findings_by_object(findings)
+        findings = self.filter_findings_by_filter_name(findings)
+
+        return findings
+
+    def get_fully_filtered_findings(self, request: HttpRequest):
+        findings = self.get_filtered_findings()
+        return self.filter_findings_by_form(request, findings)
+
+
+class ListFindings(View, BaseListFindings):
+    def get_initial_context(self, request: HttpRequest):
+        context = {
+            "filter_name": self.get_filter_name(),
+            "show_product_column": True,
+            "custom_breadcrumb": None,
+            "product_tab": None,
+            "jira_project": None,
+            "github_config": None,
+            "bulk_edit_form": FindingBulkUpdateForm(request.GET),
+            "title_words": get_words_for_field(Finding, "title"),
+            "component_words": get_words_for_field(Finding, "component_name"),
+        }
+        # Look to see if the product was used
+        if product_id := self.get_product_id():
+            product = get_object_or_404(Product, id=product_id)
+            user_has_permission_or_403(request.user, product, Permissions.Product_View)
+            context["show_product_column"] = False
+            context["product_tab"] = Product_Tab(product, title="Findings", tab="findings")
+            context["jira_project"] = jira_helper.get_jira_project(product)
+            if github_config := GITHUB_PKey.objects.filter(product=product).first():
+                context["github_config"] = github_config.git_conf_id
+        elif engagement_id := self.get_engagement_id():
+            engagement = get_object_or_404(Engagement, id=engagement_id)
+            user_has_permission_or_403(request.user, engagement, Permissions.Engagement_View)
+            context["show_product_column"] = False
+            context["product_tab"] = Product_Tab(engagement.product, title=engagement.name, tab="engagements")
+            context["jira_project"] = jira_helper.get_jira_project(engagement)
+            if github_config := GITHUB_PKey.objects.filter(product__engagement=engagement).first():
+                context["github_config"] = github_config.git_conf_id
+
+        return request, context
+
+    def get_template(self):
+        return "dojo/findings_list.html"
+
+    def add_breadcrumbs(self, request: HttpRequest, context: dict):
+        # show custom breadcrumb if user has filtered by exactly 1 endpoint
+        if "endpoints" in request.GET:
+            endpoints = request.GET.getlist("endpoints", [])
+            if len(endpoints) == 1 and endpoints[0] != '':
+                endpoint = endpoints[0]
```

Comment:
To make it clearer that we're dealing with an ID here rather than an endpoint object
```suggestion
                endpoint_id = endpoints[0]
```

Reply:
Good call. Making more changes around this to further clarify